provider/openstack/firewaller.go

```diff
@@ -70,16 +71,342 @@ type firewallerFactory struct {
 
 // GetFirewaller implements FirewallerFactory
 func (f *firewallerFactory) GetFirewaller(env environs.Environ) Firewaller {
-	return &defaultFirewaller{environ: env.(*Environ)}
+	return &switchingFirewaller{env: env.(*Environ)}
+}
+
+type switchingFirewaller struct {
```

Comment:
All the boilerplate for this wrapper is somewhat maddening. Do we actually need the switching behaviour?

I guess some of this comes under me wanting a bigger change to how firewalling is handled for openstack though.